src/main/java/de/berlin/lostberlin/repository/OrderRepository.java

```diff
@@ -1,14 +1,24 @@
 
 package de.berlin.lostberlin.repository;
 
-import de.berlin.lostberlin.model.Order;
+import de.berlin.lostberlin.model.Order.Order;
+import de.berlin.lostberlin.model.Order.OrderFullDao;
+import de.berlin.lostberlin.model.Order.OrderStatusDao;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
-import java.util.List;
 
 @Repository
 public interface OrderRepository extends JpaRepository<Order, String> {
-    List<Order> findAllByBusinessId (Long businessId);
+
+    @Query("SELECT new de.berlin.lostberlin.model.Order.OrderStatusDao(o.name, o.date, o.time, o.businessId, o.status)"+
+    "from Order o where o.orderNr = :orderNr")
+    OrderStatusDao findByOrderNr(@Param("orderNr") String orderNr);
+
+    @Query("SELECT new de.berlin.lostberlin.model.Order.OrderFullDao(o.name, o.phone, o.eMail, o.date, o.time, o.participantsNr, o.description, o.businessId, o.status, o.orderNr)"+
+    "from Order o where o.orderNr = :orderNr AND o.status = :status AND o.businessId = :businessId")
+    OrderFullDao findByOrderNrAndStatusAndBusinessId(@Param("orderNr") String orderNr, @Param("status") Enum status, @Param("businessId") Long businessId);
```

Comment:
Nitpick: since you provide sqls yourself and JPA layer does not parse your method names to create sql, you may give names a more appropriate names, for instance, getOrder(....), all params must be provided, as far as I know, should probably be annotated as NotNull.

Reply:
ok, thanks for the tipps!